src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -120,9 +120,10 @@ class CollapsingNavigationMenu extends React.Component {
 
   componentDidUpdate() {
     const { isNewSelectedPath } = this.state;
-    const currentItemPosition = this.selectedItem && this.selectedItem.current ? this.selectedItem.current.getBoundingClientRect().bottom : null;
-    const navigationMenuHeight = document.querySelector('#terra-dev-site-nav-menu').getBoundingClientRect().bottom;
-    if (isNewSelectedPath && currentItemPosition > navigationMenuHeight) {
+    const currentItemPosition = this.selectedItem && this.selectedItem.current ? this.selectedItem.current.getBoundingClientRect() : null;
+    const navigationMenuPosition = document.querySelector('#terra-dev-site-nav-menu').getBoundingClientRect();
+    if (isNewSelectedPath && currentItemPosition && navigationMenuPosition && (currentItemPosition.bottom > navigationMenuPosition.bottom
+      || currentItemPosition.top < navigationMenuPosition.top)) {
       this.selectedItem.current.scrollIntoView();
     }
   }
```

Comment:
I suggest something like this for readability.

```jsx
  componentDidUpdate() {
    const { isNewSelectedPath } = this.state;
    if(!isNewSelectedPath){
      return;
    }
    const currentItemPosition = this.selectedItem?.current ? this.selectedItem.current.getBoundingClientRect() : null;
    const navigationMenuPosition = document.querySelector('#terra-dev-site-nav-menu').getBoundingClientRect();
    if(!isNewSelectedPath || !navigationMenuPosition){
      return;
    }
    // If the current item is not visible, scroll the item into view.
    if ((currentItemPosition.bottom > navigationMenuPosition.bottom
      || currentItemPosition.top < navigationMenuPosition.top)) {
      this.selectedItem.current.scrollIntoView();
    }
  }
```

Reply:
Updated here https://github.com/cerner/terra-dev-site/pull/261/commits/f7cc5a3c33dc0374e91fefb452b76a895a8e7543